OCPP-J/src/main/java/eu/chargetime/ocpp/WebSocketListener.java

```diff
@@ -109,6 +113,42 @@ public void relay(String message) {
                 sessionFactory.createSession(new JSONCommunicator(receiver)), information);
           }
 
+          @Override
+          public ServerHandshakeBuilder onWebsocketHandshakeReceivedAsServer(WebSocket webSocket, Draft draft,
+                                                                             ClientHandshake clientHandshake) throws InvalidDataException {
+            SessionInformation information =
+                    new SessionInformation.Builder()
+                            .Identifier(clientHandshake.getResourceDescriptor())
+                            .InternetAddress(webSocket.getRemoteSocketAddress())
+                            .build();
+
+            String username = null, password = null;
+            if (clientHandshake.hasFieldValue("Authorization")) {
+              String authorization = clientHandshake.getFieldValue("Authorization");
+              if (authorization != null && authorization.toLowerCase().startsWith("basic")) {
+                // Authorization: Basic base64credentials
+                String base64Credentials = authorization.substring("Basic".length()).trim();
+                byte[] credDecoded = Base64.getDecoder().decode(base64Credentials);
+                String credentials = new String(credDecoded, StandardCharsets.UTF_8);
+                // credentials = username:password
+                final String[] values = credentials.split(":", 2);
+                if (values.length >= 1) username = values[0];
+                if (values.length >= 2) password = values[1];
```

Comment:
This won't work right: OCPP-1.6J specifically requires binary passwords to be supported, i.e. ones which might contain non-printable characters, and which cannot be converted to UTF-8 strings. You need to process the part after the ":" as a byte array. You could convert it to a hexadecimal String, though. But I would recommend to verify the length restriction first.

Reply:
You are right. Switched to byte[] instead of String, added length check as standard explicitly says password must be 20 bytes.